### PR TITLE
codeintel: create CTA for users to try out new reference panel

### LINKED ISSUE #34702: Create CTA for users to try out new reference panel
We would like to collect external feedback on our new reference panel at the same time that we've enabled it for internal dogfooding. Instead of a controlled user test, we will try out a CTA to try out a beta experience of the reference panel and collect feedback. cc: @macraig @rrhyne 

- [x] Create a new (or clean up) existing issue #30227 for external use. It should include instructions for set up, as well as instructions to provide feedback directly into the issue
- [x] Create [alert inside reference panel](https://www.figma.com/file/DZMyEJT3g2HT6HmJGNNOnW/%22Better%22-code-intel-panel?node-id=1348%3A5246) that links to the external-facing issue for reference panel 
- [x] Metrics: Can we track the count of non-sourcegraph-teammate users who have enabled the feature flag? Or would we have to track count of clicks on the CTA?

### PR DESCRIPTION
**Pull request by Numbers88s**
Closes #34702

Adds CTA to reference panel for users to try out the new version.

new-ref-panel

Adds CTA for users to provide feedback for new reference panel. The copy is temporary.

Screen Shot 2022-05-18 at 4 24 08 PM

## Test plan
Manually tested. 

## App preview:

- [Web](https://sg-web-cesar-eric-34702-reference-panel.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-scotsdfmia.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/branded/src/components/panel/views/ReferencePanelCta.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+
+import classNames from 'classnames'
+
+import { useTemporarySetting } from '@sourcegraph/shared/src/settings/temporary/useTemporarySetting'
+import { Alert, Button } from '@sourcegraph/wildcard'
+
+import styles from './ReferencePanelCta.module.scss'
+
+export const ReferencePanelCta: React.FunctionComponent = () => {
```
Comment (efritz) [inline client/branded/src/components/panel/views/ReferencePanelCta.tsx:10]: Can we get a screenshot of this component on the old ref panel for the PR description? (Do it before you set it non-draft but don't rush if we're still refactoring)

File: client/branded/src/components/panel/TabbedPanelContent.tsx [hunk 2]
```diff
@@ -141,10 +142,11 @@ export const TabbedPanelContent = React.memo<TabbedPanelContentProps>(props => {
...
         useMemo(() => haveInitialExtensionsLoaded(props.extensionsController.extHostAPI), [props.extensionsController])
     )
-
+    const [redesignedEnabled] = useTemporarySetting('codeintel.referencePanel.redesignedEnabled', false)
     const isExperimentalReferencePanelEnabled =
-        !isErrorLike(props.settingsCascade.final) &&
-        props.settingsCascade.final?.experimentalFeatures?.coolCodeIntel === true
+        (!isErrorLike(props.settingsCascade.final) &&
+            props.settingsCascade.final?.experimentalFeatures?.coolCodeIntel === true) ||
+        redesignedEnabled
```
Comment (efritz) [inline client/branded/src/components/panel/TabbedPanelContent.tsx:149]:
```suggestion
        redesignedEnabled === true
```

For consistency, we do it this way in BlobPanel.

**Review by efritz — CHANGES_REQUESTED** [review]
Things left to do:

- [ ] Create "clean" issue for users
- [ ] Add a similar dismissible alert on the **new** ref panel that links to the issue for feedback
- [ ] Have a way to transfer coolCodeIntel=true to this new temporary setting (this might only affect sg org on cloud, so maybe we just do this part by hand).

File: client/branded/src/components/panel/views/ReferencePanelCta.tsx (hunk 1 above)
Comment (Numbers88s) [reply client/branded/src/components/panel/views/ReferencePanelCta.tsx:10]: Totally. Done ✅

**Comment by rrhyne** [comment]
On the "enable the feature" banner, can we change the dismiss to the close icon?  That is our standard pattern.

**Comment by efritz** [comment]
@rrhyne Look good?

Screen Shot 2022-05-18 at 5 03 16 PM

Screen Shot 2022-05-18 at 5 03 26 PM

**Comment by efritz** [comment]
Blocked by https://github.com/sourcegraph/analytics/pull/452.

**Review by mrnugget — APPROVED** [review]

**Review by efritz — APPROVED** [review]

**Pull request merged**